CefSharp.Wpf/ChromiumWebBrowser.cs

```diff
@@ -1972,25 +1898,43 @@ protected override void OnPreviewKeyUp(KeyEventArgs e)
         /// <param name="e">The <see cref="KeyEventArgs"/> instance containing the event data.</param>
         private void OnPreviewKey(KeyEventArgs e)
         {
-            // As KeyDown and KeyUp bubble, it appears they're being handled before they get a chance to
-            // trigger the appropriate WM_ messages handled by our SourceHook, so we have to handle these extra keys here.
+            if (browser != null) 
+            {
+                var modifiers = e.GetModifiers();
+                var message = (int)(e.IsDown ? WM.KEYDOWN : WM.KEYUP);
+                var virtualKey = KeyInterop.VirtualKeyFromKey(e.Key);
+
+                browser.GetHost().SendKeyEvent(message, virtualKey, (int)modifiers);
+            }
+
             // Hooking the Tab key like this makes the tab focusing in essence work like
             // KeyboardNavigation.TabNavigation="Cycle"; you will never be able to Tab out of the web browser control.
             // We also add the condition to allow ctrl+a to work when the web browser control is put inside listbox.
+            // Prevent keyboard navigation using arrows and home and end keys
             if (e.Key == Key.Tab || e.Key == Key.Home || e.Key == Key.End || e.Key == Key.Up
                                  || e.Key == Key.Down || e.Key == Key.Left || e.Key == Key.Right
                                  || (e.Key == Key.A && Keyboard.Modifiers == ModifierKeys.Control))
             {
-                var modifiers = e.GetModifiers();
-                var message = (int)(e.IsDown ? WM.KEYDOWN : WM.KEYUP);
-                var virtualKey = KeyInterop.VirtualKeyFromKey(e.Key);
+                e.Handled = true;
+            }
+        }
 
-                if(browser != null)
+        /// <summary>
+        /// Handles the <see cref="E:PreviewTextInput" /> event.
+        /// </summary>
+        /// <param name="e">The <see cref="TextCompositionEventArgs"/> instance containing the event data.</param>
+        protected override void OnPreviewTextInput(TextCompositionEventArgs e) 
+        {
+            if (browser != null) 
```

Comment:
We need to check if (!e.Handled)` here, handle the case where an overriding implementation chooses to handle the input.